pkg/querier/cardinality_analysis_handler.go

```diff
@@ -122,6 +124,8 @@ func ActiveSeriesCardinalityHandler(d Distributor, limits *validation.Overrides)
 		}
 
 		w.Header().Set("Content-Type", "application/json")
+		w.Header().Set("Content-Length", strconv.Itoa(len(bytes)))
+		w.Header().Set(worker.ResponseStreamingEnabledHeader, "true")
```

Comment:
i'm not sure if we shouldn't deleted those two headers in `scheduler_processor.go streamResponse`. They're not really meant for the response, it's just the only way to communicate between the handler and the schedulerProcessor.

The default httpgrpc server also supports some special headers and [it removes them](https://github.com/grafana/mimir/blob/9f2a79afb8fea9cfe00ea0ed422de9ab4831f7d1/vendor/github.com/grafana/dskit/httpgrpc/server/server.go#L71-L75)

Reply:
Removed the internal-only header in 67d3110, I think `Content-Length` still makes sense here though?

Reply:
you're right. Content-Length makes sense